tests/test_api_build.py

```diff
@@ -1408,3 +1408,26 @@ def test_ignore_verify_codes(testing_config):
     # this recipe intentionally has a license error.  If ignore_verify_codes works,
     #    it will build OK.  If not, it will error out.
     api.build(recipe_dir, config=testing_config)
+
+
+def test_script_env_warnings(testing_config, recwarn):
+    recipe_dir = os.path.join(metadata_dir, '_script_env_warnings')
+    token = 'CONDA_BUILD_PYTEST_SCRIPT_ENV_TEST_TOKEN'
+    def assert_keyword(keyword):
+        messages = [str(w.message) for w in recwarn.list]
+        assert any([token in m and keyword in m for m in messages])
+        recwarn.clear()
+
+    api.build(recipe_dir, config=testing_config)
+    assert_keyword('undefined')
+
+    os.environ[token] ='SECRET'
+    try:
```

Comment:
I just think that the ``try`` here is not necessary, is it?

Reply:
I don't know pytest well enough. Just thought I'd clean up after myself.

Reply:
So, how does this work? I don't need to cleanup or I should just drop the `try - finally` part of it?

Reply:
I think that is fine then, as it is used just here should be ok
However, you can create a fixture and do the setup and teardown there. But as you are just adding and removing an env variable that is ok
:p